backend/Agent/core/agent_registry.py

```
@@ -0,0 +1,100 @@
+"""
+
+from typing import Dict, Type, List, Any
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class AgentRegistry:
+    """에이전트 자동 발견 및 등록"""
+    
+    _agents: Dict[str, Type['BaseAgent']] = {}
```

Ruff (0.14.5)

Undefined name `BaseAgent`

(F821)